### ISSUE TITLE
paypal IPN blocked by protocol anomalies rule id 960015 

### ISSUE THREAD

**Issue by fzipi** [crstest01/owasp-modsecurity-crs#193]
_Issue originally created by user beckspaced on date 2014-09-04 09:06:30.
            Link to original issue: https://github.com/SpiderLabs/owasp-modsecurity-crs/issues/193._

hello there ;-)

i'm having an issue with the paypal IPN being blocked by mod security.
here's the error log

./error_log:[Thu Sep 04 02:15:58.622136 2014] [:error] [pid 19368] [client 173.0.81.1] ModSecurity: Access denied with code 403 (phase 2). Operator EQ matched 0 at REQUEST_HEADERS. [file "/etc/apache2/mod_security2.d/modsecurity_crs_21_protocol_anomalies.conf"] [line "47"] [id "960015"] [rev "1"] [msg "Request Missing an Accept Header"] [severity "NOTICE"] [ver "OWASP_CRS/2.2.9"] [maturity "9"] [accuracy "9"] [tag "OWASP_CRS/PROTOCOL_VIOLATION/MISSING_HEADER_ACCEPT"] [tag "WASCTC/WASC-21"] [tag "OWASP_TOP_10/A7"] [tag "PCI/6.5.10"] [hostname "payment.domain.com"] [uri "/paypal/ipn.php"] [unique_id "VAevPk4vegoAAEuoNUgAAAAD"]

i'm currently disabling modsecurity for the subdomain, but isn't there a better way?

perhaps whitelist the request to ipn.php? but how can i do that?

any help would be more than welcome ;-)

thanks & greetings
becki

**Comment by fzipi** [comment]
_User beckspaced commented on date 2014-09-04 13:30:26:_

hello again,

did a bit of search in the issues posted here and found another issue with the same topic that paypal IPN being blocked:

https://github.com/SpiderLabs/owasp-modsecurity-crs/issues/123

did what was suggested by **at user rcbarnett** 

rcbarnett: You need to do an exception in a local modsecurity_crs_15_customerules.conf file for the IPN resource -

SecRule REQUEST_METHOD "POST" "chain,phase:1,t:none,nolog,pass"
SecRule REQUEST_FILENAME "@streq /ipn.php" "ctl:ruleRemoveById=960009"

but after a apache restart i receive the following error:

Sep 04 15:22:44 vs2 start_apache2[13292]: AH00526: Syntax error on line 1 of /etc/apache2/mod_security2.d/modsecurity_crs_15_custom_rules.conf:
Sep 04 15:22:44 vs2 start_apache2[13292]: ModSecurity: No action id present within the rule

any ideas on how to fix that would be great ;-)

thanks & greetings
becki

**Comment by fzipi** [comment]
_User beckspaced commented on date 2014-09-04 13:46:47:_

ups ... seems to be that id is mandatory since 2.7

add id and change the exception rule to the following did the trick:

SecRule REQUEST_METHOD "POST" "chain,id:99999,phase:1,t:none,nolog,pass"
SecRule REQUEST_FILENAME "@streq /ipn.php" "ctl:ruleRemoveById=960009"

perhaps this might help someone?

thanks & greetings
becki
